src/ufuncs.c

```diff
@@ -984,6 +875,68 @@ static void YY_Y_func(char** args, npy_intp* dimensions, npy_intp* steps, void*
 }
 static PyUFuncGenericFunction YY_Y_funcs[1] = {&YY_Y_func};
 
+/* Define the reducing geoms -> geom functions (Y_Y_reduce) */
+static void* intersection_all_data[1] = {GEOSIntersection_r};
+static void* symmetric_difference_all_data[1] = {GEOSSymDifference_r};
+static char Y_Y_reduce_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void Y_Y_reduce_func(char** args, npy_intp* dimensions, npy_intp* steps,
+                                   void* data) {
+  FuncGEOS_YY_Y* func = (FuncGEOS_YY_Y*)data;
+  GEOSGeometry* geom = NULL;
+  GEOSGeometry* ret_ptr = NULL;
+  int n_geoms;
+
+  CHECK_NO_INPLACE_OUTPUT(1);
+
+  GEOS_INIT;
+
+  GEOSGeometry** geoms = malloc(sizeof(void*) * dimensions[1]);
+  if (geoms == NULL) {
+    errstate = PGERR_NO_MALLOC;
+    goto finish;
+  }
+
+  SINGLE_COREDIM_LOOP_OUTER {
+    CHECK_SIGNALS(i);
+    if (errstate == PGERR_PYSIGNAL) {
+      goto finish;
+    }
+    n_geoms = 0;
+    SINGLE_COREDIM_LOOP_INNER {
+      if (!get_geom(*(GeometryObject**)cp1, &geom)) {
+        errstate = PGERR_NOT_A_GEOMETRY;
+        goto finish;
+      }
+      if (geom == NULL) {
+        continue;
+      }
+      geoms[n_geoms] = geom;
+      n_geoms++;
+    }
+    if (n_geoms == 0) {
+      ret_ptr = GEOSGeom_createEmptyCollection_r(ctx, 7);
+    } else {
+      ret_ptr = GEOSGeom_clone_r(ctx, geoms[0]);
+      int j;
+      for (j = 1; j < n_geoms; j++) {
+        ret_ptr = func(ctx, ret_ptr, geoms[j]);
```

Comment:
Here the old geometry should be destroyed before overwriting the `ret_ptr`.

Reply:
Good point!